subsys/logging/log_backend_net.c

```diff
@@ -0,0 +1,238 @@
+		local_addr4.sin_port = 0;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_IPV6) && server_addr.sa_family == AF_INET6) {
+		local_addr = (struct sockaddr *)&local_addr6;
+		server_addr_len = sizeof(struct sockaddr_in6);
+		local_addr6.sin6_port = 0;
+	}
+
+	if (!local_addr) {
+		DBG("Server address unknown\n");
+		return -EINVAL;
+	}
+
+	local_addr->sa_family = server_addr.sa_family;
+
+	ret = net_context_get(server_addr.sa_family, SOCK_DGRAM, IPPROTO_UDP,
+			      &ctx);
+	if (ret < 0) {
+		DBG("Cannot get context (%d)\n", ret);
+		return ret;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_HOSTNAME_ENABLE)) {
+		memcpy(hostname, net_hostname_get(), MAX_HOSTNAME_LEN);
+
+	} else if (IS_ENABLED(CONFIG_NET_IPV6) &&
+		   server_addr.sa_family == AF_INET6) {
+		const struct in6_addr *src;
+
+		src = net_if_ipv6_select_src_addr(
+			NULL, &net_sin6(&server_addr)->sin6_addr);
+		if (src) {
+			net_addr_ntop(AF_INET6, src, hostname,
+				      MAX_HOSTNAME_LEN);
+
+			net_ipaddr_copy(&local_addr6.sin6_addr, src);
+		} else {
+			goto unknown;
+		}
+
+	} else if (IS_ENABLED(CONFIG_NET_IPV4) &&
+		   server_addr.sa_family == AF_INET) {
+		struct net_if_ipv4 *ipv4;
+		struct net_if *iface;
+
+		iface = net_if_ipv4_select_src_iface(
+					&net_sin(&server_addr)->sin_addr);
+		ipv4 = iface->config.ip.ipv4;
+
+		net_ipaddr_copy(&local_addr4.sin_addr,
+				&ipv4->unicast[0].address.in_addr);
+
+		net_addr_ntop(AF_INET, &local_addr4.sin_addr, hostname,
+			      MAX_HOSTNAME_LEN);
+	} else {
+	unknown:
+		DBG("Cannot setup local context\n");
+		return -EINVAL;
+	}
+
+	ret = net_context_bind(ctx, local_addr, server_addr_len);
+	if (ret < 0) {
+		DBG("Cannot bind context (%d)\n", ret);
+		return ret;
+	}
+
+	(void)net_context_connect(ctx, &server_addr, server_addr_len,
+				  NULL, K_NO_WAIT, NULL);
+
+	/* We do not care about return value for this UDP connect call that
+	 * basically does nothing. Calling the connect is only useful so that
+	 * we can see the syslog connection in net-shell.
+	 */
+
+	net_context_setup_pools(ctx, get_tx_slab, get_data_pool);
+
+	log_output_ctx_set(&log_output, ctx);
+	log_output_hostname_set(&log_output, hostname);
+
+	return 0;
+}
+
+static void send_output(const struct log_backend *const backend,
+			struct log_msg *msg)
+{
+	if (!net_init_done) {
+		net_init_done = true;
+		if (do_net_init() < 0) {
+			net_init_done = false;
+		}
+	}
+
+	log_msg_get(msg);
+
+	log_output_msg_process(&log_output, msg,
+			       LOG_OUTPUT_FLAG_FORMAT_SYSLOG |
+			       LOG_OUTPUT_FLAG_TIMESTAMP);
+
+	log_msg_put(msg);
+}
+
+static void init_net(void)
+{
+	int ret;
+
+	net_sin(&server_addr)->sin_port = htons(514);
+
+	ret = net_ipaddr_parse(CONFIG_LOG_BACKEND_NET_SERVER,
+			       sizeof(CONFIG_LOG_BACKEND_NET_SERVER) - 1,
+			       &server_addr);
+	if (!ret) {
+		LOG_ERR("Cannot configure syslog server address");
+		return;
+	}
+
+	log_backend_deactivate(log_backend_net_get());
+}
+
+static void panic(struct log_backend const *const backend)
+{
```

Comment:
you should handle panic. I guess that in fault case sending over network is no longer possible. In that case some flag must be set here and put function should return if panic mode is enabled.

Reply:
The panic() was not implemented for other backends so I left it also out. But sure, I can add more stuff there.

Reply:
uart (and posix?) are currently synchronous, interrupt-less so no panic support was needed there.